### DISCUSSION TITLE
Constraint between foreign key and primary key not working in Dbeaver

### DISCUSSION THREAD

**Discussion by Maianh2510** [dbeaver/dbeaver#35333] [Q&A] [answered]
### Description

Hi everyone, I'm having a problem with Dbeaver. I successfully created the foreign key constraint, I chose to update casecade and delete casecade , but when I rechecked it, it didn't work.  Can you help me that why?
![2](https://github.com/user-attachments/assets/32af3283-874f-41ef-8424-b1124f603d9b)
![1](https://github.com/user-attachments/assets/db851894-f013-4098-a548-fd21d0e840b2)
![3](https://github.com/user-attachments/assets/08f7b8aa-18b0-4db9-9387-11eed5a786be)
![4](https://github.com/user-attachments/assets/53d799fd-322b-4644-86fe-da402c503df8)

### DBeaver Version

Dbeaver 24.1.5

### Operating System

window 10

**Comment by TB-HunBoka** [comment, marked as answer]
This is not a Dbeaver issue more like a question, you can ask questions like this in the discussion if you would like to: https://github.com/orgs/dbeaver/discussions

Did you turn on the PRAGMA foreign_keys setting on your sqlite db?
As far as I know it is not enabled by default and you need to enable it.
_Assuming the library is compiled with foreign key constraints enabled, it must still be enabled by the application at runtime, using the [PRAGMA foreign_keys](https://www.sqlite.org/pragma.html#pragma_foreign_keys) command._

https://docs.sqlalchemy.org/en/13/dialects/sqlite.html#foreign-key-support

**Comment by Maianh2510** [comment]
@TB-HunBoka  Thanks so much! I finished. :)

**Comment by TB-HunBoka** [comment]
Happy to help.
